vector/src/main/java/im/vector/app/features/analytics/screen/ScreenEvent.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.analytics.screen
+
+import android.os.SystemClock
+import im.vector.app.features.analytics.AnalyticsTracker
+import im.vector.app.features.analytics.plan.Screen
+import timber.log.Timber
+
+/**
+ * Track a screen display. Unique usage.
+ */
+class ScreenEvent(val screenName: Screen.ScreenName) {
```

Comment:
small nit - mainly thinking out loud, tying the start time to the object creation is a little bit hidden and means where the instance is created is implicitly tied to the duration

another way to model the behaviour could be with a separate `ScreenTracker` (which could have its own injected dependencies for clocks, trackers etc)

```kotlin
data class TimedScreen(val startTime: Long, val screenName: ScreenName)

val timedScreen: TimedScreen = screenTracker.trackEntry(screenName)

screenTracker.trackExit(trackedScreen)
```

Reply:
Yes, it's probably a little better. Happy to iterate on a dedicated PR later.